include/tc/core/cuda/cuda_mapping_options-inl.h

```diff
@@ -0,0 +1,256 @@
+/**
```

Comment:
why isn't this a .cc file? -inl would be needed for template implementations, here it looks like it should work inside .cc. .cc files are faster to compile if you later on do incremental changes to them.

Reply:
good point, will address, thanks!

Reply:
except when functions are inline, and we probably want the two-liner "fluent interface" calls to stay inline